### ISSUE TITLE
Remove the concept of private messages

### ISSUE THREAD

**Issue by tmtmtmtm** [ciudadanointeligente/write-it#436]

**Comment by mlandauer** [comment]
What's the reason for removing private messages?

**Comment by tmtmtmtm** [comment]
Deemed out of scope for this project. It's much cleaner for WriteIt to simply be a tool for sending messages in public and displaying any responses. Coping with private messages adds considerable added complexity, and muddles the core concept.

**Comment by mlandauer** [comment]
I was always under the impression that write-it was intended as an extraction of the core functionality of writetothem. It seems odd to me to suddenly say that private messages are out of scope when private messages are an important part of writing to members of parliament as demonstrated by how writetothem works.

I would prefer some public discussion of this on the poplus list because it seems to me a pretty fundamental change to what write-it is supposed to do.

**Comment by tmtmtmtm** [comment]
I never understood there to be much connection between WriteIt and WriteToThem at all. WTT is only about private messages, and only about sending them (it doesn't even try to capture any responses), it has a lot of functionality around helping people work out who to write to (which WriteIt currently does nothing with), only lets you send to one person at a time, explicitly prevents multiple messages with the same text etc. etc. 

WriteIt has always been primarily about sending messages in public (every site built with it so far has only had those) and building a permanent archive of those messages — in this regard it's much closer to Alaveteli than WriteToThem. 

I personally believe that having an individual WriteIt handle both would be quite difficult to get right without confusing users, but even if we could find a great solution to that, there simply isn't time within the remainder of the current plan to develop both sets of functionality really well, so we're focussing on public messages for now. We can certainly revisit this question later, or of course anyone else could certainly build a private-messages version of their own, but for now the goal is to do less better.

**Comment by mlandauer** [comment]
Okay, let me put it a different way. Write-it has had functionality to write private messages. Now this looks like it's going to be removed without much discussion or justification (that I can see).

Saying it makes things more complicated isn't a justification when writing private messages seems to me to be completely in scope of writing messages to politicians (especially when that functionality has been there from the start)

This makes write-it less useful and frankly makes me question whether I want to use it.

**Comment by mlandauer** [comment]
Write-it has a very strong connection to writetothem. It's mentioned in the project README! @TomSteinberg used it as an example in his talk at PoplusCon. He showed WriteToThem and then said something to the effect of "here you can make something like this with write-it".

**Comment by mlandauer** [comment]
I'll reiterate my core point. Write-it is a Poplus component and so is a community project. Removing private messaging from Write-it is a really massive change in scope and I think shouldn't be done without actively inviting feedback from the poplus community.

Creating a ticket on github (with no explanation) is not adequate. If it's @tmtmtmtm driving this change I think @tmtmtmtm should write to the poplus mailing list explaining why this change is being considered. This post should invite peoples' feedback. If need be that feedback could be directed back to this ticket.

**Comment by lfalvarez** [comment]
As we are restarting the development of WI we thought of getting rid of the private messaging part basically for simplicity. And, I think because some of the refactorings and new features we thought of focusing in the first place in public messaging. I think that given that private messages are a wanted thing, may be we do not want to remove it and wait to give it a bit of thought.

**Comment by mlandauer** [comment]
@lfalvarez thanks for the explanation. It makes sense that you should focus on getting public messages working really well if that's the main use case at the moment. The only question is about removing private messages entirely. As you suggest holding off on that, giving it a bit more thought and inviting feedback from the community, I think would be a good thing.

**Comment by lfalvarez** [comment]
Yeah it is a pretty good idea. I think that it would be a good also to say in the title of this just "Temporarily remove private messages".

**Comment by TomSteinberg** [comment]
Hi @mlandauer - thanks for asking the tough question here.

I've not been much involved in the iteration of the WriteIt spec, but I am someone who knows what it is like to over-spec something and then feel guilty and wasteful of time for not having stuck closer to a minimal feature spec before launch.

Since then I've tried to make it a policy never to criticise anyone who removes any feature before launching a tool - and in fact I try to congratulate people who have the courage and initiative to take a knife to the sacred cow, in the name of Shipping It. 

That said, you're right that a mail to the community list to suggest/announce such a change would have been a good idea. Apart from anything else, it might help smoke out which people think they might want to run sites with messages send in public vs in private.

I think renaming this ticket's also a good idea, although I'd take issue with the word 'temporarily'. I'd much rather say something like 'Change v1.0 spec so that it only covers messages sent in public'. Because after that time you never know how the project might evolve...

**Comment by dracos** [comment]
Is the complexity that it is currently optional on a per-message basis? I can see that being confusing, certainly. But if it were possible to still allow an installation-wide configuration of some sort, that would seem to make it easier to use for a WriteToThem-style site as well as a more public one. If there were a configuration variable for an installation, then that installation would really just be able to ignore some aspects of the code such as handling responses (because it wouldn't get any), and would be sending messages but not displaying them anywhere.

I agree with @mlandauer that regardless of whether the default front end is only public, that the option for messages to be considered private should remain in e.g. the database model so that reusers can potentially contribute whatever they need.

**Comment by lfalvarez** [comment]
@dracos I think that your option is very plausible and it would be understood that we want to leave the options open for future developments.
